File: test Delegation Swift/test Delegation/ViewController.m
//
//  ViewController.m
//  test Delegation
//
//  Created by Will Ge on 8/21/15.
//  Copyright (c) 2015 gewill.org. All rights reserved.
//

#import "ViewController.h"
#import "AddItemViewController.h"

@interface ViewController () <AddItemViewControllerDelegate>

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)addItem:(id)sender {
    // Initialize View Controller
    AddItemViewController *viewController = [[AddItemViewController alloc] init];
    
    // Configure View Controller
    [viewController setDelegate:self];
    
    // Present View Controller
    [self presentViewController:viewController animated:YES completion:nil];
}

- (void)viewControllerDidCancel:(AddItemViewController *)viewController {
    // Dismiss Add Item View Controller
    
}

@end

File: test Delegation Objective-C/test/ViewController.swift
//
//  ViewController.swift
//  test
//
//  Created by Will Ge on 8/21/15.
//  Copyright (c) 2015 gewill.org. All rights reserved.
//

import UIKit

class ViewController: UIViewController, AddItemViewControllerDelegate {
    func addItem(send: AnyObject) {
        // Initialize View Controller
        let viewController = AddItemViewController()
        
        // Configure View Controller
        viewController.delegate = self
        
        // Present View Controller
        presentViewController(viewController, animated: true, completion: nil)
    }
    
    func viewControllerDidCancel(viewController: AddItemViewController) {
        // Dismiss Add Item View Controller
    }
    
    func viewController(viewController: AddItemViewController, didAddItem: String) {
        
    }
    
    func viewController(viewController: AddItemViewController, validateItem: String) -> Bool {
        
        return true;
    }
}

